src/workflows/airqo_etl_utils/tests/test_bigquery_api.py

```
@@ -0,0 +1,229 @@
+    with pytest.raises(Exception):
+        bq_api.fetch_device_data_for_forecast_job(start_date_time, "train")
+
+
+def test_fetch_raw_readings_empty(mock_bigquery_client):
+    api = BigQueryApi()
+    api.client = mock_bigquery_client
+    api.client.query.return_value.result.return_value.to_dataframe.return_value = (
+        pd.DataFrame()
+    )
+    with pytest.raises(Exception) as e:
+        df = api.fetch_raw_readings()
```

Ruff (0.12.2)

Local variable `df` is assigned to but never used

Remove assignment to unused variable `df`

(F841)